src/BotWebApi/Controllers/CandidatesController.cs

```diff
@@ -20,61 +22,119 @@ public class CandidatesController : ApiController
+        public HttpResponseMessage VacanciesProgress(int candidateId)
+        {
+            HttpResponseMessage response;
+            if (_context.Candidates.Any(x => x.Id == candidateId))
+            {
+                var foundedCandidate = _context.Candidates.First(x => x.Id == candidateId);
+                response = new HttpResponseMessage()
+                {
+                    StatusCode = HttpStatusCode.OK,
+                    Content = new StringContent(JsonConvert.SerializeObject(foundedCandidate.VacanciesProgress, Formatting.Indented, new JsonSerializerSettings
+                    {
+                        
+                    }))
+                };
+            }
+            else
+            {
+                response = new HttpResponseMessage(HttpStatusCode.NotFound);
+            }
+            return response;
+        }
+
+
         [HttpDelete]
         public HttpResponseMessage Delete(int id)
         {
             HttpResponseMessage response;
             var foundedCandidate = _context.Candidates.First(x => x.Id == id);
             if(foundedCandidate != null)
-            {
-                response = new HttpResponseMessage(HttpStatusCode.Accepted);
+            { 
                 _context.Candidates.Remove(foundedCandidate);
+                if (_context.Candidates.Any(x => x != foundedCandidate)) 
+                {
+                    response = new HttpResponseMessage(HttpStatusCode.OK);
+                }
+                else
+                {
+                    response = new HttpResponseMessage(HttpStatusCode.InternalServerError);
+                }
             }
             else
             {
                 response = new HttpResponseMessage(HttpStatusCode.NoContent);
             }
-            response.Headers.Add("Access-Control-Allow-Origin", "*");
             return response;
        }
        
+        [HttpPost]
+        public HttpResponseMessage Add([FromBody]JObject entity)
+       {
+            HttpResponseMessage response = new HttpResponseMessage();
+            var newCandidate = entity.ToObject<Candidate>();
+            newCandidate.Id = _context.Candidates.Last().Id + 1;
+            _context.Candidates.Add(newCandidate);
+            response.StatusCode = HttpStatusCode.Created;
+            return response;
+        }
+
         [HttpPut]
-        public HttpResponseMessage Put(string entity)
+        public HttpResponseMessage Put(int id, [FromBody]JObject entity)
         {
-            HttpResponseMessage response;
-            var newCandidate = JsonConvert.DeserializeObject<Candidate>(entity);
-            _context.Candidates.Add(newCandidate);
-            //make connections
-            return new HttpResponseMessage(HttpStatusCode.Accepted);
+            HttpResponseMessage response = new HttpResponseMessage();
+            var changedCandidate = entity.ToObject<Candidate>();
+            var foundedCandidate = _context.Candidates.First(x => x.Id == id);
```

Comment:
this doesn't returns a null pointer, just an exception.
 so use FirstOrDefault instead.